### PR TITLE
Added item migrations tests

### PR DESCRIPTION
**Pull request by mgobec**

### CODE REVIEW

File: src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public void execute() throws MigrationException {
+        try {
+            final PreparedStatement preparedStatement =
+                    session.prepare("INSERT INTO items_by_number_external_id (id, number, external_id) VALUES (?, ?, ?);");
+
+            final List<Row> rows = session.execute(QueryBuilder.select().from("items_by_id")).all();
+            for (Row row : rows) {
```
Comment (nbozic) [inline src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java]: Maybe add fetch size to this, not that we have this many items but it shows that we know how it works and good practice.

File: src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                session.execute(preparedStatement.bind(row.getUUID("id"), row.getString("number"), row.getUUID("external_id")));
```
Comment (nbozic) [inline src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java]: Why not executeAsync?

File: src/test/java/io/smartcat/migration/migrations/schema/CreateItemByNumberAndExternalIdMigration.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package io.smartcat.migration.migrations.schema;
+
+import com.datastax.driver.core.SimpleStatement;
+import io.smartcat.migration.SchemaMigration;
+import io.smartcat.migration.exceptions.MigrationException;
+
+/**
+ * Created by matija on 3/3/16.
```
Comment (nbozic) [inline src/test/java/io/smartcat/migration/migrations/schema/CreateItemByNumberAndExternalIdMigration.java]: do we need this comment, I am not fan of it.

File: src/test/java/io/smartcat/migration/migrations/schema/CreateItemByNumberAndExternalIdMigration.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public void execute() throws MigrationException {
+        try {
+            final String statement =
+                    "CREATE TABLE IF NOT EXISTS items_by_number_external_id (" +
+                            "id uuid," +
+                            "number text," +
+                            "external_id uuid," +
+                            "PRIMARY KEY ((number, external_id))" +
+                    ") WITH COMMENT='Items by item ID';";
```
Comment (nbozic) [inline src/test/java/io/smartcat/migration/migrations/schema/CreateItemByNumberAndExternalIdMigration.java]: Comment is off

File: src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java (hunk 2 above)
Comment (mgobec) [inline src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java]: Tests will fail and its not important

File: src/test/java/io/smartcat/migration/MigrationEngineBooksTest.java [hunk 5]
```diff
@@ -25,14 +22,14 @@
 import io.smartcat.migration.migrations.data.InsertBooksMigration;
 import io.smartcat.migration.migrations.schema.AddBookGenreFieldMigration;
 
-public class MigrationEngineTest {
+public class MigrationEngineBooksTest extends BaseTest {
```
Comment (nbozic) [inline src/test/java/io/smartcat/migration/MigrationEngineBooksTest.java]: Missing after here to truncate tables as well

File: src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java (hunk 2 above)
Comment (nbozic) [inline src/test/java/io/smartcat/migration/migrations/data/PopulateItemByNumberAndExternalIdMigration.java]: ok

**Comment by nbozic** [comment]
Finished review, check my comments

**Pull request merged**